src/wasm-builder.h

```diff
@@ -741,24 +741,29 @@ class Builder {
   // Make a constant expression. This might be a wasm Const, or something
   // else of constant value like ref.null.
   Expression* makeConstantExpression(Literal value) {
-    TODO_SINGLE_COMPOUND(value.type);
-    switch (value.type.getBasic()) {
-      case Type::funcref:
-        if (!value.isNull()) {
-          return makeRefFunc(value.getFunc());
-        }
-        return makeRefNull(value.type);
+    auto type = value.type;
+    if (type.isNumber()) {
+      return makeConst(value);
+    }
+    if (type.isFunction()) {
+      if (!value.isNull()) {
+        // TODO: with typed function references we need to do more for the type
```

Comment:
What functionality does this new code add over the old code?

Reply:
Before, a function type would reach `TODO_SINGLE_COMPOUND` and trap. This does not trap, and creates a valid value (either a `ref.null` or `ref.func`).